### PR TITLE
Make jre folder platform dependent

### PR DESCRIPTION
**Pull request by chippmann**
This makes the jre folder platform dependent.

This makes it possible for a project to have multiple jre folder simultaneously on the file system. The module then automatically picks the correct one for the current platform.
The main advantage of this is that on a project with multiple people working on it, the embedded jre can be committed to the repository for each platform together with the built jar and a project member does not need to have java locally installed and JAVA_HOME set. Which is especially useful if one is working with non programmers.

**Note:** Breaking change. Needs special notes in release notes!

### CODE REVIEW

**Review by piiertho — DISMISSED** [review]
I think we need to test in exports to be sure we don't break them.

**Comment by chippmann** [comment]
I tested linux editor and export templates. Could you test windows and mac? (I don't think testing on mobile is necessary but you can if you want to :-) )

**Review by CedNaru — CHANGES_REQUESTED** [review]

**Comment by chippmann** [comment]
Tested the exports and editors for linux amd64 with the changes @CedNaru requested.
Will test on macos later today

**Comment by chippmann** [comment]
Tested editor and export on macOS as well.

If you @CedNaru or @piiertho could test on windows, that would be appreciated.

**Comment by chippmann** [comment]
Tested it on mac and windows as well in the meantime. Just graalvm is missing testing on those platforms

File: src/kotlin_editor_export_plugin.cpp [hunk 1]
```diff
@@ -40,21 +40,55 @@ void KotlinEditorExportPlugin::_export_begin(const HashSet<String>& p_features,
...
                 if (!is_arm64 && !is_x64) {
-                    add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_DIRECTORY));
+                    add_macos_plugin_file(String(RES_DIRECTORY).path_join(HOST_EMBEDDED_JRE_DIRECTORY));
                 } else {
-                    if (is_arm64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_ARM_DIRECTORY)); }
-                    if (is_x64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_AMD_DIRECTORY)); }
+                    if (is_arm64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(MACOS_EMBEDDED_JRE_ARM_DIRECTORY)); }
+                    if (is_x64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(MACOS_EMBEDDED_JRE_AMD_DIRECTORY)); }
                 }
             } else {
```
Comment (CedNaru) [inline src/kotlin_editor_export_plugin.cpp:48]: We talked about it last time. We don't support all system and architecture that godot does, so we can't assume that the generic "else" works in all cases. We should make the if extra explicit and only leave a final else to raise an error, "Platform not supported"). We can also simplify its structure a bit compared to what it is now.

**Review by CedNaru — CHANGES_REQUESTED** [review]

File: src/kotlin_editor_export_plugin.cpp (hunk 1 above)
Comment (chippmann) [reply src/kotlin_editor_export_plugin.cpp]: replaced the generic else with a proper check for universal for macos and failure in all other cases
restructured the export a bit. it should now represent what we've discussed last time IIRC

**Review by CedNaru — APPROVED** [review]

**Comment by piiertho** [comment]
I don't remember: did we test it on all platforms ?

**Comment by chippmann** [comment]
I did on all platforms except mobile and graalvm for windows and macos
Although it might be worth to test again after my last change even though i did not change much about the logic of the export

**Comment by piiertho** [comment]
I will test it on graalVM native-image for iOS.

**Comment by piiertho** [comment]
Tested on iOS

**Review by piiertho — APPROVED** [review]

**Pull request merged**